src/dependencies.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const fs = require('fs-extra')
+const path = require('path')
+const semver = require('semver')
+
+/**
+ * Determine whether GConf is a necessary dependency, given the Electron version.
+ */
+function getGConfDepends (version, distro) {
+  const gconf = distro === 'debian' ? 'libgconf2-4' : 'GConf2'
+  return semver.lt(version, '3.0.0-beta.1') ? [gconf] : []
+}
+
+/**
+ * Determine the GTK dependency based on the Electron version in use.
+ */
+function getGTKDepends (version, distro) {
+  const gtk2 = distro === 'debian' ? 'libgtk2.0-0' : 'gtk2'
+  const gtk3 = distro === 'debian' ? 'libgtk-3-0' : 'gtk3'
+  return semver.gte(version, '2.0.0-beta.1') ? gtk3 : gtk2
+}
+
+/**
+ * Determine the dependencies for the `shell.moveItemToTrash` API, based on the
+ * Electron version in use.
+ */
+function getTrashDepends (version, distro) {
+  const gvfs = distro === 'debian' ? 'gvfs-bin' : 'gvfs-client'
+  const glib = distro === 'debian' ? 'libglib2.0-bin' : 'glib2'
+  if (semver.lt(version, '1.4.1')) {
+    return gvfs
+  } else if (semver.lt(version, '1.7.2')) {
+    return `kde-cli-tools | kde-runtime | trash-cli | ${gvfs}`
+  } else {
+    return `kde-cli-tools | kde-runtime | trash-cli | ${glib} | ${gvfs}`
+  }
+}
+
+/**
+ * Determine whether libuuid is necessary, given the Electron version.
+ */
+function getUUIDDepends (version, distro) {
+  const uuid = distro === 'debian' ? 'libuuid1' : 'libuuid'
+  return semver.gte(version, '4.0.0-beta.1') ? [uuid] : []
+}
+
+/**
+ * Reads the electron version file
+ */
+function getElectronVersion (options) {
+  return fs.readFile(path.resolve(options.src, 'version'))
```

Comment:
Haven't read through this code too much but worth noting the Electron team is planning on publishing nighties to a separate package. (`electron-nightly` is the current leaning).  So this version logic should probably take that into account as a possibility

Reply:
I think that's more of a Electron Packager concern rather than `electron-installer-*`, unless `electron-nightly` (or whatever it's going to be called) isn't going to have a semver-friendly version (or it's not going to distribute the `version` file).